lib/Mojolicious/Plugin/AssetPack/Pipe/OptiPng.pm

```diff
@@ -0,0 +1,94 @@
+
+package Mojolicious::Plugin::AssetPack::Pipe::OptiPng;
+
+use Mojo::Base 'Mojolicious::Plugin::AssetPack::Pipe';
+use Mojo::Util qw( spurt slurp );
+use File::Temp qw( tempfile );
+
+sub process {
+	my ($self, $assets) = @_;
+	my $store = $self->assetpack->store;
+	my $file;
+
+	$assets->each(
+		sub {
+			my ($asset, $index) = @_;
+
+			# might also use optipng to convert BMP, GIF, PNM or TIFF to PNG
+			# but not yet
+			return unless $asset->format eq 'png';
+
+			my $attrs = $asset->TO_JSON;
+			$attrs->{key}    = 'optipng';
+			$attrs->{format} = 'png';
+
+			# already processed before?
+			return $asset->content($file)->FROM_JSON($attrs) if $file = $store->load($attrs);
+
+			my ( $content, $error );
+			{
+				my ( $fh, $file ) = tempfile();
+				spurt $store->load($asset), $file;
+
+				my ( $in, $out, $err ) = ( '', '', '' );
+				my @cmd = ( 'optipng', '-quiet', '-clobber', $file );
```

Comment:
The temp file is needed because optipng won´t take the input image data from STDIN nor write it´s output to STDOUT. It operates only on files.
The following eval is almost the same as in run() so should be equally robust; but I could indeed replace the eval with a call to run().